src/Common/ZooKeeper/ZooKeeper.h

```diff
@@ -650,7 +650,8 @@ class EphemeralNodeHolder
             return;
         try
         {
-            zookeeper.tryRemove(path);
+            if (!zookeeper.expired())
```

Comment:
But it's important to see a log message from Exception during investigation, at least we need to print something in log and increment profile event.

Reply:
1. A table has three log messages while a zookeeper session expires:

```
2023.05.25 14:40:38.327087 [ 15325 ] {} <Information> ZooKeeperClient: Finalizing session 216173496001823190. finalization_started: false, queue_finished: false, reason: 'Code: 999. Coordination::Exception: Session expired, path: xid equal to close_xid. (KEEPER_EXCEPTION) (version 23.3.2.37)'
......
2023.05.25 14:40:38.327185 [ 15380 ] {} <Warning> `700288369`.Sheet19b5d95d9aa944f58_local (ReplicatedMergeTreeRestartingThread): ZooKeeper session has expired. Switching to a new session.
......
2023.05.25 14:40:38.327400 [ 15380 ] {} <Error> zkutil::EphemeralNodeHolder::~EphemeralNodeHolder(): Cannot remove /clickhouse/tables/700288369/Sheet19b5d95d9aa944f58_local/01/replicas/saas1/is_active: Code: 999. Coordination::Exception: Session expired, path: XID overflow. (KEEPER_EXCEPTION), Stack trace (when copying this message, always include the lines below):
......
```
2. Now,` I am going to remove the third `Error` Exception message because the second `Warning `message is enough in my opinion.
3. Do we really need to add the profile event? We don't remove the path (`${zpath}/is_active`)  anymore if the session expires, so we don't need to increase the count, what do you think?

Reply:
@alesapin any comments?